### DISCUSSION TITLE
Cannot delete a deployed environment even after deleting the app that created it

### DISCUSSION THREAD

**Discussion by samwinslow** [community/community#11464] [Actions]
I have a deployed environment (Vercel) that I would like to remove from a repository. The app has been disconnected from the repository and in fact the whole Vercel account was deleted.

It is not possible to delete the environment according to [these instructions](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#deleting-an-environment). There is no section called Environments in the project settings sidebar.

If those docs are out of date, that's understandable, it just seems really strange to remove that settings option.

**Comment by minseungseon** [comment]
Hi, I'm also having the same issue. Can I get some hints on how you managed to solve this issue?

**Reply by minseungseon** [reply]
@samwinslow

**Reply by samwinslow** [reply]
Hi Minseung, I think I eventually created a new repository in GitHub which did not have Vercel connected to it, and then ran `git remote set-url <...>` to the new remote origin. It's a little bit brute-force...

**Comment by corygabrielsen** [comment]
I am also facing this issue. The docs are out of date still.

It is not possible to remove an vercel deployment Environment from a repository. Very annoying.

**Comment by alan-rep** [comment]
Same issue here

**Comment by samus-aran** [comment]
Hi everyone!

Just want to ask is the issue that you can not see "Settings" on your repo with the deployed environment or that you can not access your deployed environments through the UI? That will help us clarify the need to pass along to the internal teams.

Alternatively have you been successful on deleting a deployment via the API? Check out the [documentation here](https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#delete-a-deployment). Let us know how you get on 👍

**Reply by alan-rep** [reply]
When we go to settings we don't have the "Environments" In the left sidebar, this is an enterprise feature i guess

**Reply by nero-sensei** [reply]
@samus-aran hey, even i am having this issue. 
In the docs its said that environments can be deleted by going into repo settings and then going into the environments tab in the sidebar, but this environments sidebar tab is not present.

**Comment by phojie** [comment]
F! , THE GITHUB DOCS ARE OUT OF DATE!

**Comment by guilhermeprokisch** [comment]
I was able to deleted the environments just after delete the Vercel app and delete all deployments one by one  via api https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#delete-a-deployment

**Reply by ivanempire** [reply]
Dear GitHub,
There has to be a better way.

**Comment by ductilestudios** [comment]
One more roundabout way to go about this is to use this [delete-deployment-environment action](https://github.com/marketplace/actions/delete-deployment-environment) to delete the deployments & environment.   I was having the same problems noted here, but this worked.

I was able to create a fine grained personal access token with  Read and Write access to actions, administration, deployments, environments, repository custom properties, and workflows (not sure all those are needed but I was getting impatient) and not have to go through setting up a Github App as noted on the action readme.

**Reply by scruplelesswizard** [reply]
I was able to scope down to the following and have the action clean up the deployments and environments:
- Read access to 
  - metadata
- Read and Write access to 
  - actions
  - administration
  - deployments
  - environments

**Comment by leerob** [comment]
👋 For the Vercel folks here, you can now configure the retention period for deployments through the dashboard and CLI.

For example, canceled and errored deployments might be set to 30 days retention, while production deployments might be set to 1 year. Recently deleted deployments are shown in your project settings and can be instantly restored within 30 days of deletion.

I hope this is helpful for you. Learn more in our [documentation](https://vercel.com/docs/security/deployment-retention).

**Comment by ProgrammingLife** [comment]
Doesn't work for me. I get a message from Curl:

```
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/deployments/deployments#delete-a-deployment",
  "status": "404"
}
```

I used Fine-grained personal access token with deployment read & write permissions.

**Comment by chuanqisun** [comment]
This issue still exist even today. Why would you allow the creation of the environment but not the deletion of it via UI unless you are on some paying plan? The asymmetry makes no sense.
